mongo/collection.go

```diff
@@ -1096,7 +1192,15 @@ func (coll *Collection) Find(ctx context.Context, filter interface{},
 	return newCursorWithSession(bc, coll.registry, sess)
 }
 
-// FindOne returns up to one document that matches the model.
+// FindOne executes a find operation (https://docs.mongodb.com/manual/reference/command/find/) and returns a
+// SingleResult for one document in the collection.
+//
+// The filter parameter should be a document containing query operators and can be used to select the document to be
+// returned. It cannot be nil. An empty document (e.g. bson.D{}) will select one of the documents in the collection. If
```

Comment:
I talked about this a lot with @jyemin and @iwysiu yesterday. Even without a sort, the exact document is chosen by the ordering of the _id index on non-sharded clusters. We felt that going into this much detail wasn't necessary, especially because using an empty filter on a collection with multiple documents isn't a common use case afaik.

As for the sort option, I think that should be documented in the `options` package, not here.

Reply:
I disagree at least in part.  I think you want to be clear that if the filter and options result in a set of documents, the *first* one will be returned, not some random one.  That's a hint to users that they need to think about sort (or even skip for that matter).